codec/src/main/java/io/netty/handler/codec/MessageAggregator.java

```diff
@@ -154,9 +154,16 @@ public boolean acceptInboundMessage(Object msg) throws Exception {
     protected abstract boolean isAggregated(I msg) throws Exception;
 
     /**
-     * Returns the maximum allowed length of the aggregated message in bytes.
+     * Returns the maximum allowed length of the aggregated message in bytes. (int)
      */
-    public final long maxContentLength() {
+    public final int maxContentLength() {
```

Comment:
we cant change this as it would be API breakage

Reply:
How? If the user is passing `int`, it'll be automatically converted to long and when we cast `long` back to `int`, it'll just work. If the user is passing `long` then he should call `maxContentLength0` instead of `maxContentLength`.